presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkConfig.java

```diff
@@ -477,4 +492,27 @@ public PrestoSparkConfig setAdaptiveJoinSideSwitchingEnabled(boolean adaptiveJoi
         this.adaptiveJoinSideSwitchingEnabled = adaptiveJoinSideSwitchingEnabled;
         return this;
     }
+
+    /**
+     * Parses the catalog properties from the input string in format catalog.key=value and returns a nested map of properties.
+     */
+    public static Map<String, Map<String, String>> parseCatalogProperties(String input)
+    {
+        Map<String, String> inputMap = MAP_SPLITTER.split(nullToEmpty(input));
+
+        Map<String, Map<String, String>> result = new HashMap<>();
+
+        for (Map.Entry<String, String> entry : inputMap.entrySet()) {
```

Comment:
might want some error handling here so the user gets a helpful message if the config isn't properly formatted.